lib/src/button/button.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Input, ViewEncapsulation, EventEmitter, Output, OnInit } from "@angular/core";
+
+export type ButtonTheme = "primary" | "secondary" | "danger" | "outline-primary" | "outline-danger" | "dark" | "light" | "link";
+
+export type ButtonSize = "lg" | "md" | "sm";
+export type ButtonIconPosition = "right" | "left";
+
+export type ButtonTypes = "submit" | "button" | "reset";
```

Comment:
I'm not sure but I think there might be a proper type for this already if you check the html button>type TS definition and see if you can use that instead.